hathor/p2p/resources/mining.py

```diff
@@ -46,15 +46,18 @@ def render_POST(self, request):
 
             :rtype: bytes
         """
+        raw_data = request.content.read()
+        if raw_data is None:
+            return b'0'
```

Comment:
Why would it happen? Add a comment here explaining, please.

Reply:
I'm not sure how I ended up with that. I meant to check whether request.content is not None. I'll fix this.

Reply:
Done.

Reply:
Ok. Thanks!

Reply:
I just "remembered" how I ended up with that. When I changed to test if `request.content` is None, these tests broke: `pytest tests/resources/p2p/test_mining.py -k 'v1 and invalid_data' -s`, I'm not entirely sure why, it might have to do with this: https://docs.python.org/3/library/io.html#io.RawIOBase.read But I haven't checked if `content` has that base.

Reply:
I've kept the `if request.content is None` check because it popped up when I added annotations to the method. Which is why I thought I meant that in the first place.

Reply:
It's funny that it worked nicely before the upgrade. I feel that your changes are harmless and fix issues related to the upgrade itself.

Reply:
Actually I think that it worked before because it raised an `AttributeError` when trying to access `.decode` from the `None` returned by `request.content.read()`, but because I moved it out of the `try` block, this no longer fell to the `execpt` block and needed its own treatment.

Reply:
Got it! Better this way. Thanks for diving deep to find out what changed.